### PR TITLE
Add individual unsigned comparison enum consts in BoolTest

### PR DESCRIPTION
**Pull request by sviswa7**
Added individual unsigned comparison enum consts in BoolTest. Also fixed a bug introduced in previous commit.

### CODE REVIEW

File: src/hotspot/cpu/x86/x86.ad [hunk 1]
```diff
@@ -1227,10 +1227,12 @@ static inline Assembler::AvxVectorLen vector_length_encoding(const MachNode* use
 }
 
 static inline bool is_unsigned_booltest_pred(int bt) {
-  return  ((bt & BoolTest::unsigned_compare) == BoolTest::unsigned_compare);
+  return  (bt == BoolTest::ule ||
```
Comment (PaulSandoz) [inline src/hotspot/cpu/x86/x86.ad]: You could just test if the unsigned_compare bit is set?

File: src/hotspot/share/opto/subnode.hpp [hunk 2]
```diff
@@ -304,9 +304,10 @@ class CmpD3Node : public CmpDNode {
 // Convert condition codes to a boolean test value (0 or -1).
 // We pick the values as 3 bits; the low order 2 bits we compare against the
 // condition codes, the high bit flips the sense of the result.
+// For vector compares, additionaly, the 4th bit indicates if the compare is unsigned
 struct BoolTest {
-  enum mask { eq = 0, ne = 4, le = 5, ge = 7, lt = 3, gt = 1, overflow = 2, no_overflow = 6, never = 8, illegal = 9,
-                              unsigned_compare = 16 };
+  enum mask { eq = 0, ne = 4, le = 5, ge = 7, lt = 3, gt = 1, overflow = 2, no_overflow = 6, never = 8, illegal = 9, unsigned_compare = 16,
+              ule = unsigned_compare | le, uge = unsigned_compare | ge, ult = unsigned_compare | lt, ugt = unsigned_compare | gt };
```
Comment (PaulSandoz) [inline src/hotspot/share/opto/subnode.hpp:310]: John was suggesting that we add asserts to the existing methods, asserting that the unsigned_compare is not set. WDYT?

File: src/hotspot/cpu/x86/x86.ad (hunk 1 above)
Comment (sviswa7) [reply src/hotspot/cpu/x86/x86.ad]: Done

File: src/hotspot/share/opto/subnode.hpp (hunk 2 above)
Comment (sviswa7) [reply src/hotspot/share/opto/subnode.hpp:310]: Added assert to the constructor node.

**Comment by sviswa7** [comment]
I have addressed your review commits in the updated commit. Please let me know if anything else is missing.

**Pull request merged**